### PR TITLE
Update status codes

### PR DESCRIPTION
**Pull request by vpchung**
With this PR:

- when a POST operation is run, a status code of `201` is returned along with just the entity ID created (#43)
- when a DELETE operation is run, an empty JSON object is returned (#42)
- consolidate `4xx` status codes returned for POST operations (#51), e.g. `400` for invalid request bodies, ~~`404` for unknown query parameters,~~ `409` for values already found in database
- create new EmptyObject schema for empty JSON object

### CODE REVIEW

**Comment by tschaffter** [comment]
The PR is not passing the `lint` test

**Comment by tschaffter** [comment]
For POST, return only 400 (invalid input format or values), 201, 409 (conflict), 500 (?)

**Comment by vpchung** [comment]
@tschaffter I have updated the status codes for the POST operations, per our discussion.  The lint issues have also been addressed.

Ready for review!

File: openapi/paths/challenges.yaml [hunk 1]
```diff
@@ -11,19 +11,15 @@ post:
         schema:
           $ref: ../components/schemas/Challenge.yaml
   responses:
-    '200':
+    "201":
```
Comment (tschaffter) [inline openapi/paths/challenges.yaml:14]: We need to decide whether we go for single or double quotes. Are you following a reference for choosing the double quotes? In the NLP Sandbox I went for single quotes as used in the [openapi documentation](https://swagger.io/docs/specification/describing-responses/).

File: openapi/components/responses/Success.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+description: Successful operation
+content:
+  application/json:
+    schema:
+      $ref: ../schemas/EmptyObject.yaml
```
Comment (tschaffter) [inline openapi/components/responses/Success.yaml:5]: We do have successful responses that do not return an empty object. I would remove this schema.

Comment (tschaffter) [reply openapi/components/responses/Success.yaml:5]: In the NLP Sandbox, I don't have a named response object for success, e.g.
https://github.com/Sage-Bionetworks/rocc-schemas/blob/f09e71eda9d89fe780f5a067036fc38763e37019/openapi/paths/challenges%40%7BchallengeId%7D.yaml#L31

In order to limit the creation of new file, I would do as I do in the NLP Sandbox.

Note that the content of the `components/responses/` folder is different from the `components/schemas/<resource>CreateResponse`. The former are proper OpenAPI response object, while the latter represent objects that are used as payload for responses. I considered that naming schemas `<resource>CreateResponse` may be confused with the content of the folder `components/responses/` but I didn't find better alternatives.

File: openapi/paths/challenges@{challengeId}.yaml [hunk 3]
```diff
@@ -29,13 +27,7 @@ delete:
...
-        application/json:
-          schema:
-            $ref: ../components/schemas/Challenge.yaml
-    '403':
-      $ref: ../components/responses/Unauthorized.yaml
-    '404':
-      $ref: ../components/responses/NotFound.yaml
\ No newline at end of file
+    "200":
+      $ref: ../components/responses/Success.yaml
```
Comment (tschaffter) [inline openapi/paths/challenges@{challengeId}.yaml:31]: See above comment about success response. Instead return 200 + EmptyObject.yaml as payload.

File: openapi/paths/grants@{grantId}.yaml [hunk 4]
```diff
@@ -29,13 +27,7 @@ delete:
...
-        application/json:
-          schema:
-            $ref: ../components/schemas/Grant.yaml
-    '403':
-      $ref: ../components/responses/Unauthorized.yaml
-    '404':
-      $ref: ../components/responses/NotFound.yaml
\ No newline at end of file
+    "200":
+      $ref: ../components/responses/Success.yaml
```
Comment (tschaffter) [inline openapi/paths/grants@{grantId}.yaml:31]: Return 200 + EmptyObject as payload

File: openapi/paths/organizations.yaml [hunk 5]
```diff
@@ -17,17 +17,15 @@ post:
         schema:
           $ref: ../components/schemas/Organization.yaml
   responses:
-    '200':
+    "201":
       content:
         application/json:
           schema:
-            $ref: ../components/schemas/Organization.yaml
+            $ref: ../components/schemas/OrganizationId.yaml
```
Comment (tschaffter) [inline openapi/paths/organizations.yaml:24]: See above

File: openapi/paths/organizations@{organizationId}.yaml [hunk 6]
```diff
@@ -12,15 +12,13 @@ get:
   description: Returns the organization specified
   operationId: getOrganization
   responses:
-    '200':
+    "200":
       description: Success
       content:
         application/json:
           schema:
             $ref: ../components/schemas/Organization.yaml
```
Comment (tschaffter) [inline openapi/paths/organizations@{organizationId}.yaml:20]: Return OrganizationCreateResponse

File: openapi/paths/organizations@{organizationId}.yaml [hunk 7]
```diff
@@ -29,13 +27,7 @@ delete:
...
-        application/json:
-          schema:
-            $ref: ../components/schemas/Organization.yaml
-    '403':
-      $ref: ../components/responses/Unauthorized.yaml
-    '404':
-      $ref: ../components/responses/NotFound.yaml
\ No newline at end of file
+    "200":
+      $ref: ../components/responses/Success.yaml
```
Comment (tschaffter) [inline openapi/paths/organizations@{organizationId}.yaml:31]: See above

File: openapi/paths/persons.yaml [hunk 8]
```diff
@@ -10,19 +10,15 @@ post:
         schema:
           $ref: ../components/schemas/Person.yaml
   responses:
-    '200':
+    "201":
       content:
         application/json:
           schema:
-            $ref: ../components/schemas/Person.yaml
+            $ref: ../components/schemas/PersonId.yaml
```
Comment (tschaffter) [inline openapi/paths/persons.yaml:17]: Return PersonCreateResponse

File: openapi/paths/persons@{personId}.yaml [hunk 9]
```diff
@@ -29,13 +27,7 @@ delete:
...
-        application/json:
-          schema:
-            $ref: ../components/schemas/Person.yaml
-    '403':
-      $ref: ../components/responses/Unauthorized.yaml
-    '404':
-      $ref: ../components/responses/NotFound.yaml
\ No newline at end of file
+    "200":
+      $ref: ../components/responses/Success.yaml
```
Comment (tschaffter) [inline openapi/paths/persons@{personId}.yaml:31]: Return EmptyObject

File: openapi/paths/tags.yaml [hunk 10]
```diff
@@ -17,19 +17,15 @@ post:
         schema:
           $ref: ../components/schemas/Tag.yaml
   responses:
-    '200':
+    "201":
       content:
         application/json:
           schema:
-            $ref: ../components/schemas/Tag.yaml
+            $ref: ../components/schemas/TagId.yaml
```
Comment (tschaffter) [inline openapi/paths/tags.yaml:24]: Return TagCreateResponse

File: openapi/paths/tags@{tagId}.yaml [hunk 11]
```diff
@@ -29,13 +27,7 @@ delete:
...
-        application/json:
-          schema:
-            $ref: ../components/schemas/Tag.yaml
-    '403':
-      $ref: ../components/responses/Unauthorized.yaml
-    '404':
-      $ref: ../components/responses/NotFound.yaml
\ No newline at end of file
+    "200":
+      $ref: ../components/responses/Success.yaml
```
Comment (tschaffter) [inline openapi/paths/tags@{tagId}.yaml:31]: Return EmptyObject

File: openapi/paths/challenges.yaml (hunk 1 above)
Comment (vpchung) [reply openapi/paths/challenges.yaml:14]: My VSCode is set up to format on save, and it's currently adhering to `autopep8` standards ([pypi here](https://pypi.org/project/autopep8/)), hence the change here.

At the end of the day though, it doesn't matter which we use.  I generally use double-quotes for string literals anyway, and single-quotes for symbol-like strings, such as keys (like in the case here). Happy to disable my auto-format in that case.

**Comment by tschaffter** [comment]
@vpchung I guess you closed this PR because you want to do everything in #53. Is that correct?

File: openapi/paths/challenges.yaml (hunk 1 above)
Comment (tschaffter) [reply openapi/paths/challenges.yaml:14]: > At the end of the day though, it doesn't matter which we use. I generally use double-quotes for string literals anyway, and single-quotes for symbol-like strings, such as keys (like in the case here). 

I do the same. Yes, please give a break to your PEP8 auto-formatter on OpenAPI files. :)

**Comment by vpchung** [comment]
It closed on its own when I renamed the branch :sweat:  I will probably move forward with #53 though, since all the changes implemented here are also found in that PR!

**Comment by tschaffter** [comment]
This usually happens when you change the base branch and GitHub detects that the PR does not bring anything new to the target branch. So you are right on spot!

**Pull request closed without merging**